### PR TITLE
feat: LT-2270 - Optional document upload

### PR DESCRIPTION
**Pull request by iichr**

### CODE REVIEW

File: applications/views/documents.py [hunk 1]
```diff
@@ -46,26 +53,35 @@ def get(self, request, **kwargs):
...
 
-        logging.info(self.request)
-        data, error = add_document_data(request)
-
-        if error:
-            return form_page(request, form, extra_data={"draft_id": draft_id}, errors={"documents": [error]})
+        if (
+            request.FILES
+            or ("/end-user" in request.path and _is_application_export_type_permanent(application))
+            or "additional-document" in request.path
```
Comment (andytaylorkainos) [inline applications/views/documents.py:64]: can you add a comment above to explain th conditon? :)

Comment (iichr) [reply applications/views/documents.py:66]: Done

File: applications/helpers/check_your_answers.py [hunk 2]
```diff
@@ -413,3 +414,7 @@ def get_total_goods_value(goods: list):
 
 def _is_application_export_type_temporary(application):
     return application.get("export_type").get("key") == TEMPORARY
+
+
+def is_application_export_type_permanent(application):
+    return False if not application.get("export_type") else (application.get("export_type").get("key") == PERMANENT)
```
Comment (iichr) [inline applications/helpers/check_your_answers.py:420]: this is also called by MOD applications hence the need to check for an export type

File: applications/helpers/reverse_documents.py [hunk 3]
```diff
@@ -44,6 +48,7 @@ def document_switch(path):
         }
     elif "additional-document" in path:
         return {
+            "optional": False,
```
Comment (iichr) [inline applications/helpers/reverse_documents.py:51]: Additional documents must be provided if the section is filled in

File: applications/views/parties/end_users.py [hunk 4]
```diff
@@ -25,10 +26,10 @@ def get(self, request, **kwargs):
                     application=application,
                     editable=application["status"]["value"] == "draft",
                 ),
-                "highlight": ["Document"] if not application["end_user"]["document"] else {},
+                "highlight": ["Document"] if (is_permanent_app and not application["end_user"]["document"]) else {},
```
Comment (iichr) [inline applications/views/parties/end_users.py:29]: A document must be uploaded in end user only if the application is standard permanent

File: applications/forms/documents.py [hunk 5]
```diff
@@ -5,11 +5,14 @@
 from lite_forms.generators import confirm_form
 
 
-def attach_document_form(application_id, strings, back_link):
+def attach_document_form(application_id, strings, back_link, optional):
```
Comment (jfaracik) [inline applications/forms/documents.py:8]: Should be is_optional in the param :) `ngs, back_link, is_optional):`

File: applications/helpers/check_your_answers.py (hunk 2 above)
Comment (jfaracik) [reply applications/helpers/check_your_answers.py:420]: Could you do `return application.get("export_type", {}).get("key") == PERMANENT`

Comment (iichr) [reply applications/helpers/check_your_answers.py:420]: Not possible I'm afraid ☹️ returns None if it's not found and then the subsequent `.get` fails as unknown attribute

File: applications/forms/documents.py (hunk 5 above)
Comment (iichr) [reply applications/forms/documents.py]: Sorted 💯

File: applications/helpers/check_your_answers.py (hunk 2 above)
Comment (iichr) [reply applications/helpers/check_your_answers.py:420]: Could change to multi-line if you reckon it'd be clearer, seems too verbose though
```
export_type = application.get("export_type")
if export_type:
  return export_type.get.... == PERMANENT
return False

Comment (jfaracik) [reply applications/helpers/check_your_answers.py:420]: Nah I think it's fine the way it is 🚀

**Pull request merged**